src/PKSim.Infrastructure/Serialization/Xml/Serializers/UsedBuildingBlockXmlSerializer.cs

```
@@ -33,6 +33,9 @@ protected override void TypedDeserialize(UsedBuildingBlock usedBuildingBlock, XE
          var buildingBlockNode = usedBuildingBlockNode.Elements().First();
          var serializer = SerializerRepository.SerializerFor(buildingBlockNode);
 
+         if (serializer == null)
```

hum... that's probably some serious shit going wrong here. Or why would you accept null? If this is a valid use casse, maybe comment why?

---

This is when you have the element and you are trying to find the serializer. If the element is not a known type, then it should not throw an exception. That change is made also in OSPSuite.Serializer.

---

Or maybe we should leave this one as throwing an exception?

---

I am just very surprised by the `specificity` of this one.  Is this the only place where we retrieve a serializer liek this?

---

because heere we would have a UsedBuildingBlock WITHOUT a building block. That feels like it should NEVER happen

---

Yeah, that's right. We usually don't do this Elements.First() and then try to get the serializer. This is unique.

I guess this means that the order of the XML is important. Maybe an issue should be entered for that.

---

Why wouldn't there be 

```
Map(x => x.BuildingBlock);
```

---

ah, it's because the actual building block type is unknown I guess.

---

yeah exactly. That's definitely a bit weird
The other options would be to make sure the tool does not crash if the ref to the BB is not found

---

We can just throw the serializer not found exception as it would have done before with old ospsuite.serializer.

---

I changed it to throw the Serializer not found exception. That's what would have happened in previous OSPSuite.Serializer if a serializer can't be found from the element.